### PR TITLE
test added

### PR DESCRIPTION
**Pull request by diya2124**
test_validate_due_date_valid: Ensures that valid future and today's dates are accepted by validate_due_date.

test_validate_due_date_invalid: Verifies that a ValueError is raised for past dates in validate_due_date.

test_create_task: Confirms that create_task successfully adds a new task.

test_get_all_tasks: Checks that get_all_tasks retrieves the correct tasks from the database.

test_update_task: Verifies that update_task updates an existing task's details.

test_delete_task: Confirms that delete_task removes a task from the database.

test_get_stats: Ensures get_stats returns accurate task statistics.

Each test prints "PASSES" upon success

### CODE REVIEW

File: task_database/test_manager.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import pytest
+from datetime import datetime
+from task_manager import validate_due_date, get_all_tasks, create_task, update_task, delete_task, get_stats
+
+# --- TESTS ---
+
+def test_validate_due_date_valid():
+    """Test that validate_due_date function works with valid future and today's dates."""
+    try:
```
Comment (sreedevijain19) [inline task_database/test_manager.py]: The try/except block inside the test is unnecessary when using pytest.raises. You can directly use pytest.raises to check that the ValueError is raised.

File: task_database/test_manager.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            validate_due_date("2020-05-10")  # Past date
+    except Exception as e:
+        print(f"test_validate_due_date_invalid: FAILED - {e}")
+        raise
+    else:
+        print("test_validate_due_date_invalid: PASSES")
+
+def test_create_task():
+    """Test that create_task adds a task successfully."""
+    try:
```
Comment (sreedevijain19) [inline task_database/test_manager.py]: These tests do not validate the outcome of the operations. For example, after creating, updating, or deleting a task, you should check the database to ensure the operations were successful.
After performing create_task, update_task, or delete_task, assert that the task is present, updated, or deleted as expected.

File: task_database/test_manager.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        print(f"test_delete_task: FAILED - {e}")
+        raise
+    else:
+        print("test_delete_task: PASSES")
+
+def test_get_stats():
+    """Test that get_stats returns the correct task statistics."""
+    try:
+        stats = get_stats()
+        assert stats == [("pending", 5), ("completed", 3)]  # Expected task stats
```
Comment (sreedevijain19) [inline task_database/test_manager.py]: You are hardcoding the expected result for get_stats. This could fail if the task data changes or the database is modified.  Use fixtures / constants to set up data and then assert the expected result based on the set data

File: task_database/test_manager.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    try:
+        with pytest.raises(ValueError):
+            validate_due_date("2020-05-10")  # Past date
+    except Exception as e:
+        print(f"test_validate_due_date_invalid: FAILED - {e}")
+        raise
+    else:
+        print("test_validate_due_date_invalid: PASSES")
+
+def test_create_task():
```
Comment (sreedevijain19) [inline task_database/test_manager.py:14]: Add tests to handle edge cases like empty task descriptions, invalid dates, or invalid priority values

**Review by sreedevijain19 — APPROVED** [review]
